includes/class-dispatcher.php

```diff
@@ -169,11 +168,81 @@ public static function send_to_followers( $outbox_item_id, $batch_size = 50, $of
 		}
 	}
 
+	/**
+	 * Retry sending to followers.
+	 *
+	 * @param array $inboxes        The inboxes to retry.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @param int   $attempt        The attempt number.
+	 */
+	public static function retry_send_to_followers( $inboxes, $outbox_item_id, $attempt = 1 ) {
+		$retries = self::send_to_inboxes( $inboxes, $outbox_item_id );
+
+		// Retry failed inboxes.
+		if ( ++$attempt < 3 && ! empty( $retries ) ) {
+			self::schedule_retry( $retries, $outbox_item_id, $attempt );
+		}
+	}
+
+	/**
+	 * Send to inboxes.
+	 *
+	 * @param array $inboxes        The inboxes to notify.
+	 * @param int   $outbox_item_id The Outbox item ID.
+	 * @return array The failed inboxes.
+	 */
+	private static function send_to_inboxes( $inboxes, $outbox_item_id ) {
+		$json    = Outbox::get_activity( $outbox_item_id )->to_json();
+		$actor   = Outbox::get_actor( \get_post( $outbox_item_id ) );
+		$retries = array();
+
+		foreach ( $inboxes as $inbox ) {
+			$result = safe_remote_post( $inbox, $json, $actor->get__id() );
+
+			if ( is_wp_error( $result ) && in_array( $result->get_error_code(), self::$retry_error_codes, true ) ) {
+				$retries[] = $inbox;
```

Comment:
What do we do with the other errors? It would be nice to somehow clean up profiles that results in errors!?!

See `Followers::add_error()` and/or https://github.com/Automattic/wordpress-activitypub/blob/ef24c8bf375dbacc19c7fe505da94239b62abc9d/includes/class-scheduler.php#L127

Reply:
(Might not be the best cleanup method, but we can use that PR to improve it)

Reply:
I'm actually not sure it's super easy to do that. Since we're only looking at inboxes, not followers, it might not be possible to track that back to an individual follower, for example if the use a shared inbox.

Maybe this is a situation where we should let perfection get in the way of progress?